### PR TITLE
[receiver/mongodbreceiver] Fix flaky integration test by removing  4_2 integration test

### PR DESCRIPTION
**Pull request by JonathanWamsley**
**Description:** <Describe what has changed.>

The Mongodbreceiver is occasionally failing integration tests as shown from [source1](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20763) and [source2](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20920). These 2 errors report that integration 5_0 is emitting 0 metrics which has not occurred until #20655 got merged.  Before #20655 only integration 5_0 was running and stable for the longest time. This makes me think that this may be a resource problem.

Currently there are 4 docker integration tests running in parallel, which is excessive in my opinion. I don't think all versions(4, 4.2, 4.4 with lpu, and 5.0) are all necessary for testing. Version 4.0 and 4.2 report the same exact metric output in the golden expected4_0.yaml and expected_4.2.yaml. I therefore suggest we drop testing 4.2 and see if flakiness stops. I am not certain this will solve the flakiness problem, but it is currently my best plan.

**Link to tracking Issue:** <Issue number if applicable>
#20763 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by djaglowski** [comment]
Could we still be facing the problem described in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20654, where tests are interfering with each other?

**Comment by JonathanWamsley** [comment]
> Could we still be facing the problem described in #20654, where tests are interfering with each other?

I don't believe so. When running the test in parallel all the test case names are reporting correctly as shown in the image below. Before #20654 , 5_0 was being tested 4 times. Now all of them are being tested, but for some reason 5_0 appears to not be emitting any metrics in the 2 examples shown. The flakiness appears to be sparse, which makes me think it is a resource problem.

Screenshot 2023-04-24 at 10 34 47 AM

File: .chloggen/mongodbreceiver-fix-flaky-integration-rm-4-2-test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
```
Comment (djaglowski) [inline .chloggen/mongodbreceiver-fix-flaky-integration-rm-4-2-test.yaml:2]: Please remove the changelog entry.

**Review by djaglowski — APPROVED** [review]

**Comment by JonathanWamsley** [comment]
@djaglowski rebased and all tests are passing now

**Review by djaglowski — APPROVED** [review]

**Pull request merged**